### PR TITLE
Add application option to disable `notify_remove_create` parameter in `set_subscribe_callback` API

### PR DESCRIPTION
**Pull request by abitmore**
<del>Temporarily commented out for future improvement.</del>

Update:
Added an `"enable-subscribe-to-all"` option in `application.cpp`, default value is `false`. Force-pushed.

### CODE REVIEW

**Comment by pmconrad** [comment]
I suppose this breaks things for some API users. What's the reason for the change?

**Comment by abitmore** [comment]
I guess nobody is using this flag except probably the block chain explorers and arbitrage bots. When it's set to `true`, witness node (API server) will push every new object and ID of every removed object to the client, which IMHO can be used to DDoS public API servers.

**Comment by abitmore** [comment]
Perhaps better if have a startup option for it, disabled by default?

**Comment by pmconrad** [comment]
If it's not being used then there will no performance gain. If it is in use this change will break things.

Making it configurable would be ok IMO.

File: libraries/app/application.cpp [hunk 1]
```diff
@@ -404,6 +405,9 @@ namespace detail {
             _force_validate = true;
          }
 
+         if( _options->count("enable-subscribe-to-all") )
+            _app_options.enable_subscribe_to_all = _options->at("enable-subscribe-to-all").as<bool>();
```
Comment (abitmore) [inline libraries/app/application.cpp]: @pmconrad perhaps this `as` call need to be revised along with other changes in #714?

Comment (pmconrad) [reply libraries/app/application.cpp]: This as() is not a variant::as(), it's from boost::program_options. I don't think we need to sanitize that.

**Pull request merged**